src/encoder.rs

```diff
@@ -169,6 +169,14 @@ impl<'a, W: Write> Encoder<'a, W> {
         }
     }
 
+    pub fn new_with_info(w: W, info: Info<'a>) -> Encoder<'a, W> {
```

Comment:
Please name this `with_info` to be consistent with the "General constructors" entry in the [API guidelines](https://rust-lang.github.io/api-guidelines/naming.html#casing-conforms-to-rfc-430-c-case).